### PR TITLE
feat: show tunneling information notification when starting bot with remote skills

### LINKED ISSUE #7155: When testing a local bot connected to remote skills no guidance to configure tunnelling is shown
## Describe the bug

When testing a local bot which is connected to remote skills it's important to ensure that we detect this situation and tell the developer that this isn't going to work unless they configure tunnelling (e.g. ngrok) and update the SkillHost endpoint. Otherwise HTTP connection errors are reported and it's not clear what's going on unless you know the skill communication pattern in detail.

## Version

1.4.0-nightly.237101

## Browser

- [x] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [ ] macOS
- [x] Windows
- [ ] Ubuntu

## To Reproduce

1. Create a new Bot
2. Connect a skill (e.g. https://djtest6.azurewebsites.net/manifests/djbfskill1-2-1-manifest.json) 
3. Start Bot, observe the skills response isn't correctly retrieved, you get a 500 error reported.

## Expected behavior

- Detect that remote skills are configured in the project
- Warning when starting the project that remote tunneling will be needed in order for a locally tested bot to call remote skills
- Link to documentation or a window with a suggested ngrok command `ngrok http 3980 --host-header=localhost` (find the port that the bot is using locally!)

### PR DESCRIPTION
**Pull request by a-b-r-o-w-n**
## Description

Displays a notification when a bot is started that has remote skills connected. Only shows once per bot per session.

## Task Item

closes #7155 

## Screenshots
 
Screen Shot 2021-05-04 at 12 39 30 PM

### CODE REVIEW

File: Composer/packages/client/src/components/Notifications/TunnelingSetupNotification.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+`;
+
+const copyContainer = css`
+  margin: 0;
+  margin-bottom: 4px;
+  font-size: ${FontSizes.size16};
+  font-weight: ${FontWeights.semibold};
+`;
+
+const copyIconColor = '#005A9E';
```
Comment (hatpick) [inline Composer/packages/client/src/components/Notifications/TunnelingSetupNotification.tsx:39]: FluentTheme.palette.themeDark

File: Composer/packages/client/src/components/Notifications/TunnelingSetupNotification.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+};
+
+export const TunnelingSetupNotification: React.FC<CardProps> = (props) => {
+  const { title, data } = props;
+  const port = data?.port;
+  const command = `${getNgrok()} http ${port} --host-header=localhost`;
+
+  const copyLocationToClipboard = async () => {
+    try {
+      await window.navigator.clipboard.writeText(command);
```
Comment (hatpick) [inline Composer/packages/client/src/components/Notifications/TunnelingSetupNotification.tsx:66]: does this need permission?

File: Composer/packages/client/src/components/Notifications/TunnelingSetupNotification.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    }
+  };
+
+  return (
+    <div css={container}>
+      <h2 css={copyContainer}>{title}</h2>
+      <p css={linkContainer}>
+        {formatMessage.rich('<a>Install ngrok</a> and run the following command to continue', {
+          a: ({ children }) => (
+            <Link href="https://ngrok.com/download" rel="noopener noreferrer" target="_blank">
```
Comment (hatpick) [inline Composer/packages/client/src/components/Notifications/TunnelingSetupNotification.tsx:79]: missing a unique key

File: Composer/packages/client/src/utils/os.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (hatpick) [inline Composer/packages/client/src/utils/os.ts:1]: Check this file out:
Composer/Composer/packages/client/src/recoilModel/utils/fontUtil.ts

File: Composer/packages/types/src/shell.ts [hunk 5]
```diff
@@ -76,6 +76,7 @@ export type Notification = {
   read?: boolean;
   hidden?: boolean;
   onRenderCardContent?: ((props: Notification) => JSX.Element) | React.FC<any>;
+  data?: Record<string, unknown>;
```
Comment (hatpick) [inline Composer/packages/types/src/shell.ts:79]: I'd make this any

File: Composer/packages/client/src/components/Notifications/TunnelingSetupNotification.tsx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+          ),
+        })}
+      </p>
+      <div css={commandContainer}>
+        {command}
+        <IconButton
+          ariaLabel={formatMessage('Copy project location to clipboard')}
+          iconProps={{ iconName: 'Copy' }}
+          styles={copyIconStyles}
+          title={formatMessage('Copy project location to clipboard')}
```
Comment (hatpick) [inline Composer/packages/client/src/components/Notifications/TunnelingSetupNotification.tsx:91]: the title and aria-label dont match the purpose of the button.

File: Composer/packages/client/src/components/Notifications/TunnelingSetupNotification.tsx (hunk 2 above)
Comment (a-b-r-o-w-n) [reply Composer/packages/client/src/components/Notifications/TunnelingSetupNotification.tsx:66]: Not for write: https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API

File: Composer/packages/client/src/components/Notifications/TunnelingSetupNotification.tsx (hunk 6 above)
Comment (a-b-r-o-w-n) [reply Composer/packages/client/src/components/Notifications/TunnelingSetupNotification.tsx:91]: Huh. I havdemade that change, but it looks like it was lost!

File: Composer/packages/client/src/utils/os.ts (hunk 4 above)
Comment (a-b-r-o-w-n) [reply Composer/packages/client/src/utils/os.ts:1]: Heh. Didn't see that. I'll deprecate that and use the new function.

**Review by hatpick — APPROVED** [review]

**Review by hatpick — APPROVED** [review]

**Review by hatpick — APPROVED** [review]

**Pull request merged**